### PR TITLE
HDX-9931 - deploy ckan locally using devcontainers

### PR DESCRIPTION
**Pull request by IanHopkinson**
This PR includes changes required to deploy ckan locally into devcontainers for use with VS Code, developed on a Windows 11 machine.

The main action is in improving documentation in `.devcontainer/README.md` with some changes to the `docker/hdxckantool-ng.py` which implements the `hdxckantool`. There is a script to install the `awscli` in `.devcontainer` which is nasty (curls a file, unzips it and then installs, and my shell skills are limited) but as recommended by AWS. `pip install awscli` installs the old V1 CLI which is no longer recommended.

A bunch of JavaScript files appeared during installation, so I committed them. And there is a `less` folder which I deleted because at one point I thought it was blocking installation.

The `devcontainer/README.md` should get a developer to a point where they can run tests, and also deploy a slightly sluggish copy of the HDX Website. Reducing the docker mem_limit for the containers from 16GB each to 4GB each greatly reduced page render times (from ~2 minutes per page to 15 seconds). Possibly reducing mem_limit to 2GB will provide further improvement.

I can't add a login to HDX because recaptcha does not work on localhost.

The VS Code test runner fails during discovery, I think because the repo pins an older version of pytest/pluggy. 
https://github.com/microsoft/vscode-python/issues/23707

Next steps:
1. See if test discovery in VS Code works if we update pytest and pluggy to latest versions;
2. Add type-hinting to `hdxckantool-ng.py`;
3. Check performance of local HDX with mem_limit so to 2GB;

This PR is provided for discussion and review. I think it needs some improvement.

### CODE REVIEW

File: .devcontainer/awscli-install-run-once.sh [hunk 1]
```diff
@@ -0,0 +1,5 @@
+set -x
+apt install unzip
+curl "https://awscli.amazonaws.com/awscli-exe-linux-x86_64.zip" -o "awscliv2.zip"
+unzip awscliv2.zip
+./aws/install
```
Comment (teodorescuserban) [inline .devcontainer/awscli-install-run-once.sh]:
```suggestion
./aws/install

```

File: .vscode/settings.json [hunk 2]
```diff
@@ -0,0 +1,7 @@
+{
+    "python.testing.pytestArgs": [
+        "ckan/tests"
+    ],
+    "python.testing.unittestEnabled": false,
+    "python.testing.pytestEnabled": true
+}
```
Comment (teodorescuserban) [inline .vscode/settings.json]:
```suggestion
}

```

**Review by teodorescuserban — CHANGES_REQUESTED** [review]
- I think you need to remove all .js and .css files committed.
- The aws cli v2 installation might be perhaps better added in the Dockerfile , so you have it included already
- Please remember than the hdxckantool is used on the shared infrastructure environments and needs to be able to operate correctly with both local development (where ckan can be the superuser and can remove and create databases) as well as on AWS RDS (where there is no readily available access at superuser) - I will need more time to review the hdxckantool changes
- ending ith an empty new line is highly encouraged (even by gihub ui).

File: .devcontainer/awscli-install-run-once.sh (hunk 1 above)
Comment (IanHopkinson) [reply .devcontainer/awscli-install-run-once.sh]: Fixed in 4490493cac43fe134f839c44b25291c96a687d51

File: .vscode/settings.json (hunk 2 above)
Comment (IanHopkinson) [reply .vscode/settings.json]: Fixed in 4490493cac43fe134f839c44b25291c96a687d51

**Comment by IanHopkinson** [comment]
@teodorescuserban to install `awscli` using Docker, do I just need to add the following to `.devcontainer/Dockerfile?

`RUN /srv/ckan/.devcontainer/awscli-install-run-once.sh`

File: ckanext/reclineview/theme/public/vendor/flotr2/flotr2.js
Comment (alexandru-m-g) [inline ckanext/reclineview/theme/public/vendor/flotr2/flotr2.js:1]: I don't think that any js/css file in `ckanext/reclineview/theme/public/vendor` was an intended change. It has maybe something to do with line endings ?

I assume this and the next 2 files in the same folder do not need any change.

File: ckanext-hdx_theme/ckanext/hdx_theme/less
Comment (alexandru-m-g) [inline ckanext-hdx_theme/ckanext/hdx_theme/less:1]: There was a symlink here that seems to have been removed. We need it.

File: requirements.in
Comment (alexandru-m-g) [inline requirements.in:1]: I think that changing the requirements / dependencies should be a different PR. It directly affects how CKAN is running and it will need dedicated testing time.

**Review by alexandru-m-g — CHANGES_REQUESTED** [review]

File: ckanext-hdx_theme/ckanext/hdx_theme/less
Comment (IanHopkinson) [reply ckanext-hdx_theme/ckanext/hdx_theme/less:1]: I reverted this in [5b38bbe](https://github.com/OCHA-DAP/hdx-ckan/pull/6391/commits/5b38bbe6d61c60a232e686094e55bd94957ad9ca) - symlinks are a nightmare when passing through a Windows system, I think because Windows has a completely different model for symlinks (or none at all).

File: ckanext/reclineview/theme/public/vendor/flotr2/flotr2.js
Comment (IanHopkinson) [reply ckanext/reclineview/theme/public/vendor/flotr2/flotr2.js:1]: I reverted the various JS and CSS files in [6a9617b](https://github.com/OCHA-DAP/hdx-ckan/pull/6391/commits/6a9617b28b5213d8a5d61d67bbd3cc4ce434192b), [5b38bbe](https://github.com/OCHA-DAP/hdx-ckan/pull/6391/commits/5b38bbe6d61c60a232e686094e55bd94957ad9ca) and [2342091](https://github.com/OCHA-DAP/hdx-ckan/pull/6391/commits/23420913d4011d502ab9a5b965f94e01436bd983)

I did this with git commands like `git checkout origin/dev ckan/public/base/i18n/fr.js`

File: requirements.in
Comment (IanHopkinson) [reply requirements.in:1]: I reverted the requirements files in [cf1f681](https://github.com/OCHA-DAP/hdx-ckan/pull/6391/commits/cf1f68101da8962034fb2e9074a012008d8e3e9f) - using the `git checkout origin/dev ckan/public/base/i18n/fr.js` - seems to have resulted in some residual changes. Perhaps my checkout incantation picks up dev from when I started the branch rather than current.
